### PR TITLE
Add the rotator_verification.ipynb.

### PR DESCRIPTION
**Pull request by teweitsai**
This is the translation from the test description provided by vendor.

### CODE REVIEW

File: ttsai/rotator_verification.ipynb [hunk 1]
```diff
@@ -0,0 +1,3831 @@
+{
```
Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>How are flags generally tested?</p><p>How are the following flags are checked during the tests?</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pt2ptMoveComplete</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;newPt2PtCommand</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;moveSuccess</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;positionSet</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;initializationComplete</p><p>&nbsp;</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;stopComplete</p><p>&nbsp;</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;followingError</p><p>&nbsp;</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;positionFeedbackFault</p><p>&nbsp;</p><p>-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;rotatorOdometry_deg</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='32'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    await slew_test(rotator, 0.7, 2, 0.1)
</pre><p>oki, looks good to me</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='33'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>This is test 4.4</p><p> </p><p>For the steps in 4.3 add here annotations that will be send to the EFD and a not along the lines: "The constant speed test 4.3 will be executed using the EUI."</p><p> </p><p>How is the change of configuration parameters done e.g. for :</p><p><em>Parameters_upperPosLimit_deg</em> and <em>Parameters_RangeMaximum_deg</em></p><p> </p><p><em>Can we add a description here on how the configuration parameters got changed and where, even if it is not part of the python code itself to not forget it? </em></p><p>This is done by changing a file on GitHub right?</p><p> </p><p>Steps done using the EUI should be included in the analysis and get their own plots.</p>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='46'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Limit are at 0.5 deg at this moment right? Lets change movement to +-5 deg as suggested on page 22.</p><p> </p><p>For the next step send annotations to the EFD and mention that the next step is done by the EUI</p><ul><li>Execute a 3.5 deg/s constant speed move. The rotator should stop 1 s after hitting the 0.5 deg positive limit</li><li>Execute a 3.5 deg/s constant speed move. The rotator should not move.</li></ul><p> </p><p>Step done by the EUI should be included in the plots.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='47'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>ok</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='76'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>4.8 No tracking commands: How is these tests realised?</p><p> </p><p>I made it to look until here. I close the review for the moment so you can see my comments. </p><p>I will go on later with 4.9 to the end.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='82'/>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: <p>Some of them (such as positionSet) can not be tested because there is no related telemetry. But you can base on move() to see the result.</p><p> </p><p>Some of them (such as pt2ptMoveComplete) is inherited in the events I subscribed from EFD (if you see the implementation of ts_mtrotator, you will see the way to combine several telemetry into a single event). But the straight forward way is to generate the binary file from EUI and analyze it.</p><p> </p><p>But I think the system engineer team prefers to use the EFD.</p>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: <p>I am not sure an easy way to do this programmatically to get the EUI data from EFD. You need to hard-code the time to subscribe the EFD data.</p>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: <p>This test condition (+- 90 deg) is documented in the updated TMA test plan (and the google sheet in JIRA ticket).</p>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: <p>When there is no tracking commands in a period of time, you will see the rotation transition to Fault state and there is the related telemetry (becomes the event in CSC) to check. The Fault state is intentional.</p>

Comment (hdrass) [reply ttsai/rotator_verification.ipynb:1]: <p>Please add some text showing the mapping of internal flags to the events in the EFD</p>

Comment (hdrass) [reply ttsai/rotator_verification.ipynb:1]: <p>Please add test where we </p><ol><li>stop intentionally the tracking command from being send</li><li>read the EFD and see if the noNewtrackCmdError flag is set correctly.</li><li><br></li></ol><p>Test description says to use the steps from the last test and temove all the position and speed commands which are applied after the 6th<em> </em>second of simulation. The LSST command will repeat after this time. As the simulation takes 10 s, the error should be triggered. </p>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Add here that the setup notebook is used to bring up rotator and mount/CCW</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Please add two more positions. Can they be 50 degree away from each other?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Please add to all step messages like:</p><p>In the preamble</p><pre class="ql-syntax" spellcheck="false">start_time = datetime.now()
test_message = "Rotator New Trjectory Verification Test"
script = salobj.Controller("Script", index=10)
await asyncio.sleep(10) #wait 10 second may help with DDS problems; closing all other kernels may help too
print(f'time to start is {datetime.now() - start_time} [s]')
</pre><p>and before each test step:</p><pre class="ql-syntax" spellcheck="false">now = datetime.now()
script.log.info(f"START -- {test_message} -- LVV-T2539 -- Test step desciption -- Starting Time: {now} UTC")
</pre><p>And after each test step:</p><pre class="ql-syntax" spellcheck="false">now = datetime.now()
script.log.info(f"START -- {test_message} -- LVV-T2539 -- Test step desciption -- Stopping Time: {now} UTC")
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Here starts test 4.9 Slewing supersession. </p><p>Test looks good.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='83'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Add jerk/torque plots here and plat to show difference between actual and demanded speed, acceleration, position, etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='84'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>This is test 4.11, please add a test for 4.10 to test the following error.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='89'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Please add a test for negative minimum speed. Add plot son jerk/ torque</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='90'/>

Comment (hdrass) [inline ttsai/rotator_verification.ipynb:1]: <p>Please add the test numbers from the test Testplan. This is test 4.12.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/72/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='96'/>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: This following error is from the Copley drive (not the CCW). You could not do this unless you intentionally break the Copley drive hardware (and we may need to send it back to vendor to fix it). Therefore, it is better to use the Simulink model for this test only.

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: The test 4.12 will be removed based on Ander (we had moved it to the previous test case).

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: This is not included in the Ander's Simulink test case. This script is to translate what he has in the Simulink test.

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: This is not included in the Ander's Simulink test case. This script is to translate what he has in the Simulink test.

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: This will be updated in DM-34077. Thanks!

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: It is not easy to fine-tune that position. The existed test case can test the slewing supersession already. Thanks!

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: <p>This is included in <strong>Test Case (tracking / slewing sequence test) </strong>and <strong>Test Case (tracking / slewing with multiple stops)</strong> already. Thanks!</p><p> </p>

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: Updated in the commit: 137c4fd to f4b262b already. There is the case-sensitive error in your folder and I could not reach the LSP at summit and Tucson Teststand to rewrite the history. Stay like this at this time. Thanks!

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: Updated in the commit: 137c4fd to f4b262b already. There is the case-sensitive error in your folder and I could not reach the LSP at summit and Tucson Teststand to rewrite the history. Stay like this at this time. Thanks!

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: Updated in the commit: 137c4fd to f4b262b already. There is the case-sensitive error in your folder and I could not reach the LSP at summit and Tucson Teststand to rewrite the history. Stay like this at this time. Thanks!

Comment (teweitsai) [reply ttsai/rotator_verification.ipynb:1]: Updated in the commit: 137c4fd to f4b262b already. There is the case-sensitive error in your folder and I could not reach the LSP at summit and Tucson Teststand to rewrite the history. Stay like this at this time. Thanks!

**Review by hdrass — APPROVED** [review]
